File: HCF.py
# take input of two number and Calculate the HCF 

a = int(input('Ente the first number '))
b = int(input("Enter the second nubmer :"))
l1 = []
for i in  range(1,a//2+1 or b//2+1):
    if( a%i== 0 and b%i == 0 ):
        l1.append(i)


print(l1)

File: HCF.c
#include<stdio.h>
int main()
{
    int n,a,b,m;
    printf("\n++++++++++++++++++++++++++++++++++++\n Calculate the HCF of two number \n ++++++++++++++++++++++++++++++++\n ");
    printf("Enter the  both number :");
    scanf("%d%d",&n,&m);

    for(int i = 1;i<= n/2+1|| i<= m/2+1;i++)
    {
        if(n%i==0 && m%i ==0)
        printf("%d, ",i);
    }
    

    return 0;
}